File: STT.js
const speech = require('@google-cloud/speech')
const service_js = require('./key/service_js.js')
const fs = require('fs')
const _ = require('lodash')

// 임시 변수 설정
const title = 'test'
const filepath = './stt_test_0.mp3'

const STT = (filepath) => {
  // 서비스 연결
  service_js.connect()
  
  // 클라이언트 설정
  const client = new speech.SpeechClient();

  // 오디오 파일 열기
  const file = fs.readFileSync(filepath)
  const audioBytes = file.toString('base64')
  const audio = {
    content: audioBytes,
  }

  // 변환 설정
  const config = {
    encoding: 'WEBM_OPUS',
    sampleRateHertz: 24000,
    languageCode: 'ko-KR',
  }

  // 변환 요청
  const request = {
    audio,
    config,
  }

  // 작업 시행
  client
  .recognize(request)
  .then((data) => {
    const results = _.get(data[0], 'results', []);
    const transcription = results
      .map(result => result.alternatives[0].transcript)
      .join('\n');
    console.log(`Transcription: ${transcription}`);
  })
  .catch(err => {
    console.error('ERROR:', err);
  })
}

// 함수 실행
STT(filepath)

File: STT.py
from key import service
from google.cloud import speech


def speech_to_text(audio_path):

    client = speech.SpeechClient()

    with open(audio_path, "rb") as audio_file:
        content = audio_file.read()

    audio = speech.RecognitionAudio(content=content)

    config = speech.RecognitionConfig(
        encoding=speech.RecognitionConfig.AudioEncoding.WEBM_OPUS,
        sample_rate_hertz=16000,
        language_code="ko-KR",
    )

    operation = client.long_running_recognize(config=config, audio=audio)

    print("Waiting for operation to complete...")
    response = operation.result(timeout=90)
    
    for result in response.results:
        print(u"Transcript: {}".format(result.alternatives[0].transcript))
        print("Confidence: {}".format(result.alternatives[0].confidence))


# 구글 인증 및 클라이언트 등록
service.connect()

# 경로 설정
audio_path = "stt_test_0.mp3"

# 프로세스 실행
speech_to_text(audio_path=audio_path)